library/test/src/cli.rs

```diff
@@ -281,6 +281,10 @@ fn parse_opts_impl(matches: getopts::Matches) -> OptRes {
 
     let options = Options::new().display_output(matches.opt_present("show-output"));
 
+    if logfile.is_some() && !format.is_programmatic() {
```

Comment:
Can you explain why `format.is_programmatic()` is needed? Since we are now printing to stderr the warning shouldn't mess up the contents of stdout if it is redirected. Or is it intentional that --logfile is not deprecated with these output formats?

Reply:
Yes, if we've moved to stderr, then its not really needed.